### PR TITLE
TASK: Avoid heavily use of controller context and make `Neos.Ui.NodeInfo` internal

### PR DESCRIPTION
**Pull request by mhsdesign**
_If_ for some reason `Neos.Ui.NodeInfo` was used in consumer fusion land it must be migrated.
For migration please copy the parts of the `NodeInfo` helper to your own helper.
Note that the helper is sorely an internal detail of the user interface.

**What I did**

**How I did it**

**How to verify it**

### CODE REVIEW

File: Classes/Fusion/Helper/NodeInfoHelper.php [hunk 1]
```diff
@@ -30,6 +30,7 @@
 use Neos\Neos\Utility\NodeTypeWithFallbackProvider;
 
 /**
+ * @internal
  * @Flow\Scope("singleton")
  * @todo EEL helpers are still to be declared as internal
```
Comment (grebaldi) [inline Classes/Fusion/Helper/NodeInfoHelper.php:35]:
```suggestion
```

This should go, since `NodeInfoHelper` is marked `@internal` now in L33.

**Review by grebaldi — APPROVED** [review]
Hi @mhsdesign,

nice work :) A left a tiny nitpicky comment, but the rest LGTM :+1:

File: Classes/Fusion/Helper/NodeInfoHelper.php (hunk 1 above)
Comment (grebaldi) [reply Classes/Fusion/Helper/NodeInfoHelper.php:35]: **UPDATE:** Just saw that #3711 contains all changes of #3703 and removes the `@todo` annotation. Maybe #3711 can just supersede #3703, so #3703 can be closed? wdyt?

Comment (mhsdesign) [reply Classes/Fusion/Helper/NodeInfoHelper.php:35]: a i forgot how i did it, but yes this pr is basically step one and later we followup with https://github.com/neos/neos-ui/pull/3711 so ill merge it first ;)
Thanks for looking so deep into it :D

**Pull request merged**